### PR TITLE
Cache: Support Jenga values, add promoted message, restore per-rule hinting.

### PR DESCRIPTION
**Pull request by mefyl**

### CODE REVIEW

File: src/cache/client.ml [hunk 1]
```diff
@@ -45,20 +45,37 @@ let make ?finally ?duplication_mode ~command_handler () =
...
-          Result.Error (errf [ Pp.textf "invalid address: %s" addr ]) )
-      | None -> Result.Error (errf [ Pp.textf "invalid port: %s" port ]) )
-    | _ -> Result.Error (errf [ Pp.textf "invalid endpoint: %s" port ])
-  in
-  let fd = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-  let* _ =
-    Result.try_with (fun () -> Unix.connect fd (Unix.ADDR_INET (addr, port)))
+  let* fd =
+    match
+      ( String.drop_prefix ~prefix:"unix:" endpoint
```
Comment (rgrinberg) [inline src/cache/client.ml:50]: Perhaps this should be `unix://` for consistency?

File: test/blackbox-tests/test-cases/dune-cache/missing-data/stat.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+stat="stat"
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/dune-cache/missing-data/stat.sh:2]: we don't use this hack anymore. Try using `$ dune_cmd stat`

File: src/cache/client.ml (hunk 1 above)
Comment (mefyl) [reply src/cache/client.ml:50]: `//` in a uri introduces a hostname; I think we do want `unix:/some/path`, which is scheme `unix` and path `/some/path`.

**Comment by mefyl** [comment]
This prototype is superseded by the upcoming JaneStreet implementation of the distributed cache.

**Pull request closed without merging**